### PR TITLE
Add option.extendDefaultFields

### PR DESCRIPTION
**Pull request by jeremija**
Hi there,

Thank you for this module - I find it very useful.

This functionality allows modification of the default fields stored in the session table. I needed to query sessions per user to enable easier revocation of existing sessions, so I don't have to parse all data in the table to find just a few sessions that might belong to a certain user.

Let me know if you have any questions / comments.

--

The method could be used to store additional data from session to the session
table, for example userId of the user.

The main reason I needed this functionality is to be able to select all
sessions by certain user. For example, I added `userId` string to
session model, and I want that field to be read from session.

The way to do it is to implement an extendDefaultFields function:

```javascript
var Session = sequelize.define('Session', {
    sid: {
      type: Sequelize.STRING,
      primaryKey: true
    },
    userId: Sequelize.STRING,
    expires: Sequelize.DATE,
    data: Sequelize.STRING(50000)
});

function extend(defaults, session) {
    return {
        data: defaults.data,
        expires: defaults.expires,
        userId: session.userId
    };
}

store = new SessionStore({db: db, table: Session, extendDefaultFields: extend});
```

and now `userId` will be stored in the database.

### CODE REVIEW

File: lib/connect-session-sequelize.js [hunk 1]
```diff
@@ -29,7 +29,10 @@ module.exports = function SequelizeSessionInit(Store) {
     }
 
     var key;
-    this.options = defaultOptions;
+    this.options = {};
+    for (key in defaultOptions) {
+      this.options[key] = defaultOptions[key];
+    }
```
Comment (mweibel) [inline lib/connect-session-sequelize.js]: why this change?

**Comment by mweibel** [comment]
I think that's a good change. Could you address my code review comment and also add something to the README to explain how to do that?

File: lib/connect-session-sequelize.js (hunk 1 above)
Comment (jeremija) [inline lib/connect-session-sequelize.js]: Because otherwise the `defaultOptions` object would be shared between instances as `this.options`.Should someone, for whatever reason, use two instances of `SequelizeStore`, they wouldn't be able to specify two different `extendDefaultFields` functions.

Now that I think of it, I didn't notice there was a check below for `if key !== 'db'`. I will push an update which uses only a single iteration over `defautObjects`.

**Comment by jeremija** [comment]
Thanks! I just pushed two changes (avoided the two loops and updated the `README.md` file).

File: lib/connect-session-sequelize.js [hunk 2]
```diff
@@ -23,16 +23,14 @@ util.inherits(SequelizeStoreException, Error);
 
 module.exports = function SequelizeSessionInit(Store) {
   function SequelizeStore(options) {
-    options = options || {};
+    this.options = options = options || {};
```
Comment (mweibel) [inline lib/connect-session-sequelize.js]: simpler would be: ` this.options = options || {};` - then you'd just use everywhere in this constructor `this.options` instead of `options`.

File: lib/connect-session-sequelize.js [hunk 3]
```diff
@@ -23,16 +23,14 @@ util.inherits(SequelizeStoreException, Error);
...
       throw new SequelizeStoreException('Database connection is required');
     }
 
-    var key;
-    this.options = defaultOptions;
-    for (key in options) {
-      if (key !== 'db') {
-        this.options[key] = options[key];
+    for (var key in defaultOptions) {
+      if (key in options === false) {
```
Comment (mweibel) [inline lib/connect-session-sequelize.js]: `if (key in options === false) {`? does that work? Never seen such syntax in JS

File: lib/connect-session-sequelize.js (hunk 2 above)
Comment (jeremija) [inline lib/connect-session-sequelize.js]: i can change it, but is it really worth it? this would just make the constructor harder to read because of all the `this.` prefixes.

File: lib/connect-session-sequelize.js (hunk 3 above)
Comment (jeremija) [inline lib/connect-session-sequelize.js]: yeah `key in options` would return `true` if `options[key]` is set, regardless if the value is truthy or falsy. it also returns `true` when a prototype of the object has this value, thus one could argue that it might be better to use `options.hasOwnProperty(key)`, in case one wouldn't want to the values' of an object's prototype.

we could also use underscore's or lodash's `_.defaults(options, defaultOptions)`.

Comment (mweibel) [inline lib/connect-session-sequelize.js]: ah, yeah. true ;) ok no is fine 👍
